test/periodic/CMakeLists.txt

```diff
@@ -12,7 +12,28 @@ include(${ROOT_USE_FILE})
 ROOT_GENERATE_DICTIONARY(G__NdbDict ${headers} LINKDEF NdbLinkDef.h)
 
 #---Create a shared library with geneated dictionary
+
 add_library(NdbDict SHARED ${DBsources} G__NdbDict.cxx)
+
+# Do not add -Dname_EXPORTS to the command-line when building files in this
+# target. Doing so is actively harmful for the modules build because it
+# creates extra module variants, and not useful because we don't use these
+# macros.
+set_target_properties(${NdbDict} PROPERTIES DEFINE_SYMBOL "")
+if (ROOT_cxxmodules_FOUND)
+  # G__NdbDict.cxx includes XSElementDlg.h which requires header files from
+  # module Graf. When building Graf we build Hist->MathCore->VecCore. We fail on
+  # building VecCore because it has to be built only by the build system from
+  # bottom up (because of the specific -I flags).
+  #
+  # Ideally, this test should pick up the modules from module cache and it must
+  # not rebuild anything. Unfortunately the module Graf is built in a
+  # -DUSE_ROOT_ERROR context. Adding this -DUSE_ROOT_ERROR will allow us to
+  # reuse the module cache. The real fix is to remove the -DUSE_ROOT_ERROR from
```

Comment:
If there is a "real fix", shouldn't we do that instead?

Reply:
Yes, we should do that but this is not possible at the moment.